core/templates/pages/release-coordinator-page/services/release-coordinator-backend-api.service.spec.ts

```diff
@@ -0,0 +1,361 @@
+      imports: [HttpClientTestingModule]
+    });
+
+    rcbas = TestBed.get(ReleaseCoordinatorBackendApiService);
+    httpTestingController = TestBed.get(HttpTestingController);
+    csrfService = TestBed.get(CsrfTokenService);
+    successHandler = jasmine.createSpy('success');
+    failHandler = jasmine.createSpy('fail');
+    jobsData = {
+      oneOffJobStatusSummaries:
+       jobsDataBackendResponse.one_off_job_status_summaries.map(
+         JobStatusSummary.createFromBackendDict),
+      humanReadableCurrentTime:
+       jobsDataBackendResponse.human_readable_current_time,
+      auditJobStatusSummaries:
+       jobsDataBackendResponse.audit_job_status_summaries.map(
+         JobStatusSummary.createFromBackendDict),
+      unfinishedJobData: jobsDataBackendResponse.unfinished_job_data.map(
+        Job.createFromBackendDict),
+      recentJobData: jobsDataBackendResponse.recent_job_data.map(
+        Job.createFromBackendDict),
+      continuousComputationsData:
+       jobsDataBackendResponse.continuous_computations_data.map(
+         ComputationData.createFromBackendDict)
+    };
+
+    spyOn(csrfService, 'getTokenAsync').and.callFake(async() => {
+      return Promise.resolve('sample-csrf-token');
+    });
+  });
+
+  afterEach(() => {
+    httpTestingController.verify();
+  });
+
+  it('should fetch the jobs data', fakeAsync(() => {
+    rcbas.getJobsDataAsync().then((adminData) => {
+      expect(adminData).toEqual(jobsData);
+    });
+
+    let req = httpTestingController.expectOne('/jobshandler');
+    expect(req.request.method).toEqual('GET');
+    req.flush(jobsDataBackendResponse);
+
+    flushMicrotasks();
+  }));
+
+  it('should use the rejection handler if the backend request failed.',
+    fakeAsync(() => {
+      rcbas.getJobsDataAsync().then(successHandler, failHandler);
+
+      var req = httpTestingController.expectOne(
+        '/jobshandler');
+      expect(req.request.method).toEqual('GET');
+
+      req.flush({
+        error: 'Some error in the backend.'
+      }, {
+        status: 500, statusText: 'Internal Server Error'
+      });
+      flushMicrotasks();
+
+      expect(successHandler).not.toHaveBeenCalled();
+      expect(failHandler).toHaveBeenCalledWith('Some error in the backend.');
+    })
+  );
+
+  // Test cases for Admin Jobs Tab.
+  it('should request to start a new job when calling startNewJobAsync',
+    fakeAsync(() => {
+      let jobType = 'ActivityContributorsSummaryOneOffJob';
+      let payload = {
+        action: 'start_new_job',
+        job_type: jobType
+      };
+      rcbas.startNewJobAsync(jobType).then(successHandler, failHandler);
+
+      let req = httpTestingController.expectOne('/jobshandler');
+      expect(req.request.method).toEqual('POST');
+      expect(req.request.body).toEqual(payload);
+      req.flush(200);
+      flushMicrotasks();
+
+      expect(successHandler).toHaveBeenCalled();
+      expect(failHandler).not.toHaveBeenCalled();
+    }
+    ));
```

Comment:
Should we do? Since they are intended in the same way?
```suggestion
    }));
```

Reply:
These tests are mostly copied from other spec files and they had this pattern and I see this in many places. Anyway, I've fixed it as you have suggested! :)

Reply:
Done!